processing/src/main/java/io/druid/segment/filter/BoundFilter.java

```diff
@@ -59,6 +61,11 @@ public boolean apply(String input)
                   if (input == null) {
                     return false;
                   }
+
+                  if (extractionFn != null) {
+                    input = extractionFn.apply(input);
```

Comment:
`Strings.nullToEmpty(extractionFn.apply(input))` (although I still think this would make more sense in DPF)